### DISCUSSION TITLE
Making tidy maps of Dominant Soil Parent Material

### DISCUSSION THREAD

**Discussion by kevinwolz** [ncss-tech/soilDB#322] [Q&A] [answered]
I have always failed to produce a clean map of Dominant Soil Parent Materials because the `copm` table in SSURGO is kind of a mess, at least to my untrained eye. It seems that there is a lot of missing data, or at least a non-uniform approach to formatting the data:
![image](https://github.com/ncss-tech/soilDB/assets/11711566/979686f0-6d56-4499-a0a5-4483e8590014)

However, I have recently come across the Dominant Soil Parent Materials displayed on Darrel Schulze's [SoilExplorer.net](http://soilexplorer.net/). This map has beautiful, contiguous data grouped into a handful of tidy categories: 
![image](https://github.com/ncss-tech/soilDB/assets/11711566/19aa907f-71f8-490b-ad5a-240a336165c2)

Any idea how these maps were created??

**Comment by dylanbeaudette** [comment, marked as answer]
It is a good question and something that is being "addressed" using regionally-varying strategies. Unfortunately, it is currently a low priority. Several issues to content with:

  1.  possibly multiple parent material kind / origin per component (colluvium from A over residuum from B)
  2.  possibly multiple parent material kind / origin per map unit
  3.  regional and temporal variability in terminology and degree of specificity (granitoid rocks, granite proper, granodiorite, etc.)

Our component geomorphic data have similar issues.

Simplification and correlation are on-going but there is a key missing piece: a unified / simplified legend that can be applied nationally and stored at the map unit level.

Darrel Schultz and his coauthors have put in hundreds of hours manually simplifying, correlating, and referencing a semi-unified legend across part of CONUS. Their work exists outside of SSURGO, applied manually and with (some?) pattern matching. Part of our cooperative agreement with Darrel includes "feeding back" their work into SSURGO updates.

If I were you, I'd start small and bring in (at least) a soil scientist / geomorphologist to assist with the correlation.

**Reply by kevinwolz** [reply]
Thank you so much for the quick response @dylanbeaudette! This is fascinating to learn about, and I hope that Darrel et al continue to get the funding and momentum they need to keep it up. In the meantime, I think I will avoid the complexity and wait until clearer skies emerge.

**Reply by brownag** [reply]
The "component parent material group name" that you have in your table result is a combination of entries from copm. Arguably it is what is intended for the user to see, and it is used primarily for narrative reports rather than a "category" for downstream analysis. It is common for the "calculated" group names based on the exact values of the copm child records to be wordsmithed for clarity and to remove unneeded repetition. 

As Dylan mentioned there is not a simplified national legend at the "group name" level, or a standard set of rules on types of simplifications or changes made to names, but choice lists for kind and origin are standard, and can be coupled with optional modifiers. It should be expected to have these types of many:1 relationships in "copm" table. 

Since it wasn't mentioned, and I'm not sure you've tried it, I want to point out you can get dominant condition component parent material group names with, for example: `get_SDA_pmgroupname(areasymbols = "CA041", method = "dominant condition", simplify = FALSE, miscellaneous_areas = TRUE)` which will give output 1:1 with mukey. The output could be decent starting point for creating simplified legends, you would need to write patterns or seek specific terms of interest. Also note that the default `simplify = TRUE` attempts to provide a very broad simplification of parent material kinds that may warrant further review. That logic was borrowed from an older SDA query and has not been substantially revised for the soilDB implementation.
